src/modules/UserSettings/components/AccountSecurity/AccountSecurity.jsx

```diff
@@ -0,0 +1,239 @@
+            },
+        }));
+
+        return isValid;
+    }
+
+    passwordValidation() {
+        const { password, retypePassword, serverErrors } = this.state;
+
+        let feedbackText = "";
+        if (!password || !retypePassword) {
+            feedbackText = feedbackMessages.password.empty;
+        } else if (password !== retypePassword) {
+            feedbackText = feedbackMessages.password.mismatch;
+        }
+
+        let isValid = !feedbackText;
+        if (serverErrors.password && serverErrors.password.length) {
+            isValid = false;
+        }
+
+        this.setState((prevState) => ({
+            fieldsValidation: {
+                ...prevState.fieldsValidation,
+                password: isValid,
+            },
+            fieldsFeedback: {
+                ...prevState.fieldsFeedback,
+                password: feedbackText,
+            },
+        }));
+
+        return isValid;
+    }
+
+    checkButtonAvailability() {
+        const { fieldsValidation } = this.state;
+        return Object.values(fieldsValidation).some((elem) => !elem);
+    }
+
+    render() {
+        const {
+            oldPassword,
+            password,
+            retypePassword,
+            fieldsFeedback,
+            serverErrors,
+
+            showSucceededMessage,
+        } = this.state;
+
+        return (
+            <>
+                <Form onSubmit={this.onSubmit}>
+                    <Form.Group controlId="oldPassword">
+                        <Form.Label>Old password</Form.Label>
+                        <Input
+                            ref={this.fieldRefs.oldPassword}
+                            type="password"
+                            name="oldPassword"
+                            onChange={this.onFieldChange}
+                            validationFunc={this.oldPasswordValidation}
+                            value={oldPassword}
+                        />
+                        <Feedback type="invalid">
+                            { serverErrors.old_password ? (
+                                serverErrors.old_password
+                            ) : (
+                                fieldsFeedback.oldPassword
+                            )}
+                        </Feedback>
+                    </Form.Group>
+
+                    <Form.Group controlId="password">
+                        <Form.Label>New password</Form.Label>
+                        <Input
+                            ref={this.fieldRefs.password}
+                            type="password"
+                            name="password"
+                            onChange={this.onPasswordChange}
+                            validationFunc={this.passwordValidation}
+                            value={password}
+                        />
+                    </Form.Group>
+
+                    <Form.Group controlId="password">
+                        <Form.Label>New password</Form.Label>
```

Comment:
Let's use here `Confirm new password` by analogy with the form

![image](https://user-images.githubusercontent.com/5358223/94437149-0aa95300-01a6-11eb-86e5-b3a083f5651b.png)